lib/sycamore/sycamore/llms/openai.py

```diff
@@ -276,7 +318,8 @@ def __init__(
                 client_wrapper.api_key = api_key
 
         self.client_wrapper = client_wrapper
-        self._client = self.client_wrapper.get_client()
+        self._client: OpenAIClient = self.client_wrapper.get_client()
```

Comment:
If we mark get_client and get_async_client with @functools.cache then we can get rid of both of these local variables and only calculate the client or async_client if they are used.

Reply:
changed this!